packages/contracts-bedrock/contracts/L2/L1Block.sol

```diff
@@ -23,7 +23,7 @@ contract L1Block is Semver {
         0xdcb9291dfcbfcfdf70cc580ec1c1835768ebd0ed4a668ccb9e9e6167f533e655;
 
     /// @notice The max count of block hashes to store.
-    uint16 public constant BLOCK_HASHES_SIZE = 50000;
+    uint32 public constant BLOCK_HASHES_SIZE = 2**16; // 65536
```

Comment:
Lets use the 2**16-1 (65635) value in order to fit into uint16 and not spill into uint32